druid/src/widget/slider.rs

```diff
@@ -68,7 +88,25 @@ impl Slider {
         let scalar = ((mouse_x + self.x_offset - knob_width / 2.) / (slider_width - knob_width))
             .max(0.0)
             .min(1.0);
-        self.min + scalar * (self.max - self.min)
+        let mut value = self.min + scalar * (self.max - self.min);
+        if let Some(step) = self.step {
+            let max_step_value =
+                (((value - self.min) / step).floor() * step + self.min).min(self.max);
```

Comment:
looks like a typo
```suggestion
                (((self.max - self.min) / step).floor() * step + self.min).min(self.max);
```
also I don't think .min(self.max) is needed now